store/sqlstore/store.go

```diff
@@ -403,6 +428,42 @@ func (ss *SqlStore) TotalMasterDbConnections() int {
 	return ss.GetMaster().Db.Stats().OpenConnections
 }
 
+// ReplicaLagAbs queries all the replica databases to get the absolute replica lag value
+// and updates the Prometheus metric with it.
+func (ss *SqlStore) ReplicaLagAbs() {
+	for i, item := range ss.settings.ReplicaLagSettings {
+		if *item.QueryAbsoluteLag == "" {
+			continue
+		}
+		var binDiff int
+		var node string
+		err := ss.replicaLagHandles[i].QueryRow(*item.QueryAbsoluteLag).Scan(&node, &binDiff)
+		if err != nil {
+			mlog.Warn("ReplicaLagAbs Query returned an error", mlog.Err(err))
+			return
+		}
```

Comment:
What's the reason for logging the error instead of returning it?

Reply:
Hmm .. yeah I thought about it. You mean logging the error in the enterprise metrics handling code? That might work. I just wasn't sure if it was elegant or not. See that PR and let me know if you are okay with it or not?

Reply:
I guess the final result would be the same but it just feels better to leave the caller decide what to do with it. Logging on the metrics callback sounds good to me.

Reply:
Sounds good then. I will change.